Sources/libcggen/BCCGGenerator.swift

```diff
@@ -0,0 +1,330 @@
+      return byteCommand(24, intent)
+    case let .globalAlpha(alpha):
+      return byteCommand(25, alpha)
+    case let .strokeColor(color):
+      return byteCommand(26, color)
+    case let .fillColor(color):
+      return byteCommand(27, color)
+    case let .linearGradient(name, options):
+      return byteCommand(
+        28,
+        gradientsIds[name]!,
+        options.startPoint,
+        options.endPoint,
+        options.endPoint
+      )
+    case let .radialGradient(name, options):
+      return byteCommand(
+        29,
+        gradientsIds[name]!,
+        options.startCenter,
+        options.startRadius,
+        options.endCenter,
+        options.endRadius,
+        options.options
+      )
+    case let .linearGradientInlined(gradient, options):
+      return byteCommand(
+        30,
+        gradient,
+        options.startPoint,
+        options.endPoint,
+        options.options
+      )
+    case let .radialGradientInlined(gradient, options):
+      return byteCommand(
+        31,
+        gradient,
+        options.startCenter,
+        options.startRadius,
+        options.endCenter,
+        options.endRadius,
+        options.options
+      )
+    case let .subrouteWithName(name):
+      return byteCommand(32, subroutesIds[name]!)
+    case let .shadow(shadow):
+      return byteCommand(33, shadow)
+    case let .blendMode(mode):
+      return byteCommand(34, mode)
+    case .beginTransparencyLayer:
+      return byteCommand(35)
+    case .endTransparencyLayer:
+      return byteCommand(36)
+    case let .composite(steps):
+      return generateSteps(steps: steps)
+    case .endPath, .fillColorSpace, .strokeColorSpace:
+      return []
+    }
+  }
+}
+
+extension DrawRoute: ByteCodable {
+  var byteCode: [UInt8] {
+    generateGradients(gradients: gradients) +
+      generateSubroutes(subroutes: subroutes) + generateSteps(steps: steps)
+  }
+}
+
+private var gradientsIds: [String: UInt32] = [:]
+
+private func generateGradients(gradients: [String: Gradient]) -> [UInt8] {
+  var res = UInt32(gradients.count).byteCode
+  for gradient in gradients {
+    let counter = UInt32(gradientsIds.count)
+    gradientsIds[gradient.key] = counter
+    res += counter.byteCode + gradient.value.byteCode
+  }
+  return res
+}
+
+private var subroutesIds: [String: UInt32] = [:]
+
+private func generateSubroutes(subroutes: [String: DrawRoute]) -> [UInt8] {
+  var res = UInt32(subroutes.count).byteCode
+  for subroute in subroutes {
+    let counter = UInt32(subroutesIds.count)
+    subroutesIds[subroute.key] = counter
+    let subrouteBytecode = subroute.value.byteCode
+    res += counter.byteCode + UInt32(subrouteBytecode.count)
+      .byteCode + subrouteBytecode
+  }
+  return res
+}
+
+struct BCCGGenerator: CoreGraphicsGenerator {
+  let headerImportPath: String
+
+  func filePreamble() -> String {
+    """
+    #import <CoreGraphics/CoreGraphics.h>
+    #import "\(headerImportPath)"
+    @import BCRunner;
```

Comment:
Instead of importing `BCRunner` we can just declare a function somewhere in the beginning:
```
void runBytecode(CGContextRef ctx, const char* bytecode, size_t size);
```